zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/AuthInfo.java

```diff
@@ -73,25 +97,19 @@ public DirectBuffer toDirectBuffer() {
   }
 
   public Map<String, Object> toDecodedMap() {
-    switch (getFormat()) {
-      case JWT -> {
-        final String jwtToken = getAuthData();
-        return Authorization.jwtDecoder(jwtToken)
-            .withClaim(Authorization.AUTHORIZED_TENANTS)
-            .decode();
-      }
-      default -> {
-        return Map.of(
-            Authorization.AUTHORIZED_TENANTS, List.of(TenantOwned.DEFAULT_TENANT_IDENTIFIER));
-      }
+    if (claimsProp.hasValue()) {
```

Comment:
❓ This method is still needed as we need to ensure that older records are still correctly processed, am I right ?

Reply:
Yes, we need the backwards compatibility :+1: 

:x: Using `hasValue` this way should not work as far as I can tell. `hasValue` always returns true when the property has a default value, which `DocumentProperty` always does.
I'd suggest that we instead use the `format` property to switch between decoding the JWT or just using the claims document.